runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -906,78 +906,195 @@ allocateWriteBarrierInternalPointerRegister(TR::CodeGenerator * cg, TR::Node * s
+   /* ====== VO argumentSelectorReg,argumentSelectorReg,compareResultReg     (a > b) || (a == NaN) ====== */
+   generateVRRcInstruction(cg, TR::InstOpCode::VO, node, argumentSelectorReg, argumentSelectorReg, compareResultReg, 0, 0, 0);
 
-   /* ====== For Max: WFTCIDB V17,V0,X'800'     a == +0 ====== */
+   /* ====== For Max: WFTCIDB compareResultReg,operand1,X'800'     a == +0 ====== */
    if(isMaxOp)
     {
-       generateVRIeInstruction(cg, TR::InstOpCode::VFTCI, node, v17, v0, 0x800, 8, 3);
+       generateVRIeInstruction(cg, TR::InstOpCode::VFTCI, node, compareResultReg, operand1, 0x800, 8, typeMask);
     }
-   /* ====== For Min: WFTCIDB V17,V0,X'400'     a == -0 ====== */
+   /* ====== For Min: WFTCIDB compareResultReg,operand1,X'400'     a == -0 ====== */
    else
     {
-       generateVRIeInstruction(cg, TR::InstOpCode::VFTCI, node, v17, v0, 0x400, 8, 3);
+       generateVRIeInstruction(cg, TR::InstOpCode::VFTCI, node, compareResultReg, operand1, 0x400, 8, typeMask);
     }
-   /* ====== WFTCIDB V18,V2,X'C00'       b == 0 ====== */
-   generateVRIeInstruction(cg, TR::InstOpCode::VFTCI, node, v18, v2, 0xC00, 8, 3);
+   /* ====== WFTCIDB zeroCheckReg,operand2,X'C00'       b == 0 ====== */
+   generateVRIeInstruction(cg, TR::InstOpCode::VFTCI, node, zeroCheckReg, operand2, 0xC00, 8, typeMask);
 
-   /* ====== VN V17,V17,V18     (a == -0) && (b == 0) ====== */
-   generateVRRcInstruction(cg, TR::InstOpCode::VN, node, v17, v17, v18, 0, 0, 0);
+   /* ====== VN compareResultReg,compareResultReg,zeroCheckReg     (a == -0) && (b == 0) ====== */
+   generateVRRcInstruction(cg, TR::InstOpCode::VN, node, compareResultReg, compareResultReg, zeroCheckReg, 0, 0, 0);
 
-   /* ====== VO V16,V16,V17     (a >= b) || (a == NaN) || ((a == -0) && (b == 0)) ====== */
-   generateVRRcInstruction(cg, TR::InstOpCode::VO, node, v16, v16, v17, 0, 0, 0);
+   /* ====== VO argumentSelectorReg,argumentSelectorReg,compareResultReg     (a >= b) || (a == NaN) || ((a == -0) && (b == 0)) ====== */
+   generateVRRcInstruction(cg, TR::InstOpCode::VO, node, argumentSelectorReg, argumentSelectorReg, compareResultReg, 0, 0, 0);
 
-   /* ====== VSEL V0,V0,V2,V16 ====== */
-   generateVRReInstruction(cg, TR::InstOpCode::VSEL, node, v0, v0, v2, v16);
+   /* ====== VSEL operand1,operand1,operand2,argumentSelectorReg ====== */
+   generateVRReInstruction(cg, TR::InstOpCode::VSEL, node, operand1, operand1, operand2, argumentSelectorReg);
 
    /* ===================== Deallocating Registers  ===================== */
-   cg->stopUsingRegister(v2);
-   cg->stopUsingRegister(v16);
-   cg->stopUsingRegister(v17);
+   cg->stopUsingRegister(operand2);
+   cg->stopUsingRegister(selectFirstArgReg);
+   cg->stopUsingRegister(compareResultReg);
    cg->stopUsingRegister(v18);
 
-   node->setRegister(v0);
+   node->setRegister(operand1);
 
    cg->decReferenceCount(node->getFirstChild());
    cg->decReferenceCount(node->getSecondChild());
 
    return node->getRegister();
```

Comment:
we set operand1 as node's register just couple of line ago, instead of getting register from node, we can return operand1 here.